numpy/lib/_iotools.py

```diff
@@ -26,10 +28,9 @@ def _decode_line(line, encoding=None):
 
     """
     if type(line) is bytes:
-        if encoding is None:
-            line = line.decode('latin1')
-        else:
-            line = line.decode(encoding)
+        if not encoding:
```

Comment:
```suggestion
        if encoding is None:
```
Otherwise this mishandles the empty string (which should error)